### ISSUE TITLE
Start-PSBootstrap failing on Windows 10

### ISSUE THREAD

**Issue by BurtHarris** [PowerShell/PowerShell#4000]
From a freshly cloned repository, I'm trying to build Powershell on Windows 10 using PowerShell Core 6.0.0-beta.   It fails.

It seems to work using the Windows 10 distributed PowerShell.exe.

Steps to reproduce
------------------
From repository root:

```powershell
import-module .\build.psm1
Start-PSBootstrap
```

Expected behavior
-----------------
bootstrap completes, as under desktop version of PowerShell:
```text
Installing PowerShell build dependencies
C:
dotnet-install: Downloading https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0-preview1-005952/dotnet-sdk-2.0.0-preview1-005952-win-x64.zip
dotnet-install: Downloading https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0-preview1-005952/dotnet-dev-win-x64.2.0.0-preview1-005952.zip
dotnet-install: Extracting zip from https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0-preview1-005952/dotnet-dev-win-x64.2.0.0-preview1-005952.zip
dotnet-install: Adding to current process PATH: "C:\Users\Burt_\AppData\Local\Microsoft\dotnet\". Note: This change will not be visible if PowerShell was run as a child process.
dotnet-install: Installation finished
Chocolatey is already installed. Skipping installation.
Cmake is already installed. Skipping installation.
Windows 10 SDK present. Skipping installation.
```

Actual behavior
---------------

```none
Installing PowerShell build dependencies
C:
dotnet-install: Downloading https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0-preview1-005952/dotnet-sdk-2.0.0-preview1-005952-win-x64.zip
dotnet-install: Downloading https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0-preview1-005952/dotnet-dev-win-x64.2.0.0-preview1-005952.zip
Exception calling "IsBypassed" with "1" argument(s): "Operation is not supported on this platform."
At C:\code\PowerShell\tools\dotnet-install.ps1:170 char:20
+ ...          if($DefaultProxy -and (-not $DefaultProxy.IsBypassed($Uri))) ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : PlatformNotSupportedException
```

Environment data
----------------
Latest commit 6bb12c9df5afa0a2cbbcef1289b2f253875ed50d.

There are no HTTP proxies in my environment.  I'm behind a modern home NAT router.

```text
Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
BuildVersion                   3.0.0.0
CLRVersion
GitCommitId                    v6.0.0-beta.2
OS                             Microsoft Windows 10.0.15063
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

**Comment by anmenaga** [comment]
This was discussed today.
Error derives from some .NET APIs missing on CoreCLR.
Currently Start-PSBootstrap has to be run in full Windows PowerShell.

Need to either:
- when it is launched on PowerShell Core on Windows, add a warning/error to Start-PSBootstrap saying to use Windows PowerShell.
- when it is launched on PowerShell Core on Windows, implicitly launch Windows PowerShell and run the necessary stuff in there.

**Comment by daxian-dbw** [comment]
This should be addressed by #4144. See the changes in `build.psm1` at [here](https://github.com/PowerShell/PowerShell/pull/4144/files#diff-ed2f07121a08595cc7655dda069a5ee8R1052)
